python/cuml/decomposition/pca.pyx

```diff
@@ -325,7 +330,15 @@ class PCA(Base):
 
     def _build_params(self, n_rows, n_cols):
         cpdef paramsPCA *params = new paramsPCA()
-        params.n_components = self.n_components
+        if self.n_components is None:
+            warnings.warn(
```

Comment:
I would particularly appreciate feedback on how I'm doing the deprecation warning here. Is this consistent with other deprecation messages? Too verbose? Not enough detail?

Reply:
This would not be a deprecation warning since nothing is being deprecated but the behavior is changing. Since that's the case, then using the cuml.common.logger would be the standard way we do it now: https://github.com/rapidsai/cuml/blob/964cd1172478ed36074baba45352e0e490674f9e/python/cuml/tsa/arima.pyx#L470

As a matter of fact, migrating old warnings to use the logging functionality should be done. I saw that John tagged you in #2604, so that might be good to tackle as part of solving that issue actually

Reply:
Added a note to #2604 to remind myself to look at that as I'm working on that issue. I've updated the warning to make use of cuml.common.logger. Thanks!